id: PYSEC-2017-25
details: XML External Entity (XXE) vulnerability in PySAML2 4.4.0 and earlier allows
  remote attackers to read arbitrary files via a crafted SAML XML request or response.
affected:
- package:
    name: pysaml2
    ecosystem: PyPI
    purl: pkg:pypi:pysaml2
  ranges:
  - type: GIT
    repo: https://github.com/rohe/pysaml2
    events:
    - introduced: "0"
    - fixed: 6e09a25d9b4b7aa7a506853210a9a14100b8bc9b
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 4.5.0
  versions:
  - 0.4.3
  - 1.0.1
  - 1.0.2
  - 1.0.3
  - 1.1.0
  - 2.0.0
  - 2.1.0
  - 2.2.0
  - 2.3.0
  - 2.4.0
  - 3.0.0
  - 3.0.2
  - 4.0.0
  - 4.0.1
  - 4.0.2
  - 4.0.3
  - 4.0.4
  - 4.0.5
  - 4.0.5rc1
  - 4.1.0
  - 4.2.0
  - 4.3.0
  - 4.4.0
references:
- type: WEB
  url: https://github.com/rohe/pysaml2/pull/379
- type: REPORT
  url: https://github.com/rohe/pysaml2/issues/366
- type: FIX
  url: https://github.com/rohe/pysaml2/commit/6e09a25d9b4b7aa7a506853210a9a14100b8bc9b
- type: WEB
  url: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=850716
- type: WEB
  url: http://www.openwall.com/lists/oss-security/2017/01/19/5
- type: ADVISORY
  url: http://www.debian.org/security/2017/dsa-3759
- type: WEB
  url: http://www.securityfocus.com/bid/97692
- type: ADVISORY
  url: https://access.redhat.com/errata/RHSA-2017:0938
- type: ADVISORY
  url: https://access.redhat.com/errata/RHSA-2017:0937
- type: ADVISORY
  url: https://access.redhat.com/errata/RHSA-2017:0936
aliases:
- CVE-2016-10149
modified: "2021-07-05T00:01:25.139700Z"
published: "2017-03-24T14:59:00Z"
